pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -170,7 +181,7 @@ func (v *templateBaseValue) setTemplateValue(rsrc interface{}, rsrclabels map[st
 	labels.Merge(master.Labels, serviceLabels, rsrclabels)
 
 	// Set the cdap.container label. It is for service selector to route correctly
-	name := fmt.Sprintf("cdap-%s-%s", master.Name, strings.ToLower(string(serviceType)))
+	name := getResourceName(master, strings.ToLower(string(serviceType)))
```

Comment:
Since k8s requires resource name to be all lower case, it's good to move the `ToLower` inside the function.

https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names